ldm/invoke/restoration/codeformer.py

```diff
@@ -33,9 +38,20 @@ def process(self, image, strength, device, seed=None, fidelity=0.75):
             
             cf_class = CodeFormer
             
-            cf = cf_class(dim_embd=512, codebook_size=1024, n_head=8, n_layers=9, connect_list=['32', '64', '128', '256']).to(device)
-            
-            checkpoint_path = load_file_from_url(url=pretrained_model_url, model_dir=os.path.abspath('ldm/invoke/restoration/codeformer/weights'), progress=True)
+            cf = cf_class(
+                dim_embd=512,
+                codebook_size=1024,
+                n_head=8,
+                n_layers=9,
+                connect_list=['32', '64', '128', '256']
+            ).to(device)
+
+            # note that this file should already be downloaded and cached at
+            # this point
+            checkpoint_path = load_file_from_url(url=pretrained_model_url,
+                                                 model_dir=os.path.abspath(os.path.dirname(self.model_path)),
```

Comment:
Should this be using `normpath`? Should probably standardize on either `abspath` or `normpath` (I'm not sure which is more appropriate).

Reply:
i checked the docs and could not say, but instinctively i would want to find out why self.model_path is not already in a well-known format (that would make either normpath or abspath redundant)

Reply:
Again, normpath() is mostly cosmetic, and would only make a difference if we were doing a string-based comparison of two paths. But I'd use os.path.samefile() for that.

Reply:
> i checked the docs and could not say, but instinctively i would want to find out why self.model_path is not already in a well-known format (that would make normpath and abspath redundant)

Since the user is free to edit the models.yaml file, they might write `./models/ldm/stable_diffusion/etc` or `models/ldm/stable_diffusion/etc`, or `/home/fred/InvokeAI/models/ldm/stable_diffusion/etc` Whatever they write, `abspath()` will turn it into an absolute path, whereas `normpath()` will get rid of redundant dots and path components. Usually they do the same thing, e.g. `/foo/./././bar/../bar` => `/foo/bar`. The difference is the handling of relative pathnames. `abspath()` will take on the current working directory, while `normpath()` won't. 

In this case I was calling out to a third party function and I didn't know for sure how it treats relative pathnames, so I made the path absolute to avoid any ambiguity.